src/alignment.js

```diff
@@ -90,15 +90,22 @@ var alignWithDOM = function(nodeName, key, statics) {
       if (process.env.NODE_ENV !== 'production') {
         assertKeyedTagMatches(existingNode, nodeName, key);
       }
+
       matchingNode = existingNode;
     } else {
       matchingNode = createNode(walker.doc, nodeName, key, statics);
+
       if (key) {
         registerChild(parent, key, matchingNode);
       }
     }
 
-    parent.insertBefore(matchingNode, currentNode);
+    if (currentNode && getData(currentNode).key) {
+      parent.replaceChild(matchingNode, currentNode);
```

Comment:
This'll create a subtle memory leak with `keyMap`.


```js
patch(container, function(condition) {
  if (condition) {
    elementVoid('div', 'first');
  }
  elementVoid('div', 'second');
}
```

`first` will never be removed from the `keyMap` since `lastVisitedChild` is `second` and there's nothing else after it in the parent.